### PR TITLE
Feature/uninteruppted filtering

### LINKED ISSUE #101: As a potential gamer, I want to be able to filter without clicking submit, so that I get a better search flow
- [x] Make the filter update without clicking submit
- [x] Reload the answers when filter is updated

### PR DESCRIPTION
**Pull request by Supergamer1337**
## Key Features
- Filtering is automatic! No need to presss submit.
- Different filters are cached, so no need to wait if you go back to a previous filter selection.
- General improvments to client side code.
- Closes #101 

## Known Issues
- Sometimes when emptying one of the number fields, react dev tools spams "NaN cannot be parsed" in the console.
- Really ugly useFilter hook, but I used it to clean up the component code slightly. Prefer it over the alternative.

### CODE REVIEW

File: frontend/src/components/GamesList/GamesList.tsx [hunk 1]
```diff
@@ -1,107 +1,50 @@
...
+				setPlaytimeMin={filter.setPlaytimeMin}
+				setPlayerCount={filter.setPlayerCount}
+				setOwner={filter.setOwner}
+				platform={filter.full.platform}
+				releaseAfter={filter.full.releaseAfter}
+				releaseBefore={filter.full.releaseBefore}
+				playtimeMax={filter.full.playtimeMax}
+				playtimeMin={filter.full.playtimeMin}
+				playerCount={filter.full.playerCount}
+				owner={filter.full.owner}
```
[the remark this answers is not in the document]
Comment (x183) [reply frontend/src/components/GamesList/GamesList.tsx:30]: Yes, I think that'd be the more logical thing to do

[the remark this answers is not in the document]
Comment (Supergamer1337) [reply frontend/src/components/GamesList/GamesList.tsx]: Done!

File: frontend/src/hooks/api/useAddRating.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { useMutation } from '@tanstack/react-query';
+import axios from 'axios';
+
+export const useAddRating = () => {
+	return useMutation({
```
Comment (Oscariremma) [inline frontend/src/hooks/api/useAddRating.ts:5]: Shouldn't this also have something like this to invalidate the game query?
```typescript
{
	onSuccess: () => {
		queryClient.invalidateQueries([
			'siteAdmin',
			'organization',
			props.orgId
		]);
	}
}
````
Example stolen from `OrganizationMemberListEntry.tsx`

The same applies to all of the mutaions.

File: backend/src/routers/gameRouter.ts [hunk 3]
```diff
@@ -1,33 +1,81 @@
...
+		search: z.string().min(1).max(500).optional(),
+		platform: z.string().min(1).optional(),
+		releaseBefore: z.string().datetime().optional(), // ISO date string
+		releaseAfter: z.string().datetime().optional(), // ISO date string
+		playtimeMin: z.string().min(1).regex(isInt, intMessage).optional(),
+		playtimeMax: z.string().min(1).regex(isInt, intMessage).optional(),
+		playerCount: z
+			.string()
+			.min(1)
+			.max(2000)
```
Comment (Oscariremma) [inline backend/src/routers/gameRouter.ts:38]: Why does only `playerCount` have a max number of chars? And if we have a max I then a number with 2000 chars is quite big...

**Review by Oscariremma — CHANGES_REQUESTED** [review]

**Review by GAsplund — CHANGES_REQUESTED** [review]
Seems to run as intended overall and the code cleanup looks nice. One thing I noticed is that games don't disappear once you specify a player count that's larger than the maximum players field, but I feel that maybe it should?

File: backend/src/routers/gameRouter.ts (hunk 3 above)
Comment (Supergamer1337) [reply backend/src/routers/gameRouter.ts:38]: That was probably a result of me just chaning the type to string 😅 Will fix. Yeah 2000 is a bit big. Probably more reasonable to have like 4 or something

**Comment by Supergamer1337** [comment]
> Seems to run as intended overall and the code cleanup looks nice. One thing I noticed is that games don't disappear once you specify a player count that's larger than the maximum players field, but I feel that maybe that should be the case?

Fixed!

**Review by GAsplund — APPROVED** [review]
I'm happy. LGTM!

**Review by Oscariremma — CHANGES_REQUESTED** [review]
Looks good but I found a bug.
If you enter a date and then press clear in the date dialog the request will fail with a Invalid date error

**Comment by Supergamer1337** [comment]
> Looks good but I found a bug.
> If you enter a date and then press clear in the date dialog the request will fail with a Invalid date error

I should've already fixed that bug (that's what isValidDate is for), but I'll look into it. I had to go real quick, so I should have something in a about an hour or so

**Review by Oscariremma — APPROVED** [review]
LGTM

**Pull request merged**